src/printer.rs

```diff
@@ -393,6 +393,7 @@ impl<'a> Printer for InteractivePrinter<'a> {
                     .map_err(|_| "Invalid UTF-16BE")?,
                 _ => String::from_utf8_lossy(line_buffer).to_string(),
             }
+            .replace('\u{feff}', "") // Removing BOM from all non-loop-through output
```

Comment:
This will call `.replace` on every single line in the output, right?
It will also replace the "zero width no-break space" character everywhere in the output, not just at the beginning.

Do we really need/want that? Doesn't this add a runtime overhead?

I would have expected that it is enough to call `.strip_prefix('\u{feff}')` on the very first line of the output.

https://doc.rust-lang.org/stable/std/string/struct.String.html#method.strip_prefix

Reply:
If it's expected to only appear at the first line, and always at the beginning of a line (without any other strange unicode characters before it perhaps), `strip_prefix` should be enough. The only issue I can think of then would be if we have two concatenated files, both with a BOM at the beginning. Will the BOM be stripped from them individually if we were to strip it in `print_line`?

I agree that calling `.replace` on every line might have too big of a hit on performance. I just wasn't sure if it was enough to only strip from the first line. I'm also not familiar enough with file encodings to know if the BOM always appears as the very first character.

Reply:
BOM is now only stripped when `line_number` is 1 and only from the very start of the line using `strip_prefix`.

Reply:
> The only issue I can think of then would be if we have two concatenated files, both with a BOM at the beginning

I'd say we can safely ignore this use case for now. I *think* that would be valid in terms of the Unicode standard. But it's certainly not the recommended way of designing a Unicode text file.